src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFModel.java

```diff
@@ -0,0 +1,239 @@
+     * if indicator == REF_MINOR
+     * same as ALT_MINOR case but with f <--> 1 - f
+     * if indicator == OUTLIER log {pi * a!r!/(n+1)!}
+     * where alpha = mu*beta and n = a + r
+     */
+    public static double hetLogLikelihood(final AFParameters parameters, final AllelicCount count, final AFIndicator indicator) {
+        final double mu = parameters.getMeanBias();
+        final double beta = parameters.getBiasRate();
+        final double pi = parameters.getpOutlier();
+        final int a = count.getAltReadCount();
+        final int r = count.getRefReadCount();
+
+        final double alpha = mu * beta;
+
+        if (indicator == AFIndicator.OUTLIER) {
+            return log(pi) + logFactorial(a) + logFactorial(r) - logFactorial(a + r + 1);
+        } else {
+            final double f = indicator == AFIndicator.ALT_MINOR ? parameters.getMinorFraction() : 1 - parameters.getMinorFraction();
+            final int n = a + r;
+            final double w = (1 - f) * (a - alpha + 1) + beta * f;
+            final double lambda0 = (sqrt(w * w + 4 * beta * f * (1 - f) * (r + alpha - 1)) - w) / (2 * beta * (1 - f));
+            final double kappa = n * square(1 - f) / square(f + (1 - f) * lambda0) - (r + alpha - 1) / square(lambda0);
+            final double rho = 1 - kappa * square(lambda0);
+            final double tau = -kappa * lambda0;
+            final double logc = alpha*log(beta) - Gamma.logGamma(alpha) + a * log(f) + r * log(1 - f)
+                    + (r + alpha - rho) * log(lambda0) + (tau - beta) * lambda0 - n * log(f + (1 - f) * lambda0);
+
+            return log((1 - pi) / 2) + logc + Gamma.logGamma(rho) - rho * log(tau);
+        }
+    }
+
+    /**
+     * the log likelihood summed (marginalized) over indicator states, which we use in the fully collapsed model
+     * in which latent variables (bias and indicator) are marginalized out
+     *
+     * @param parameters allele fraction parameters at this het site
+     * @param count AllelicCount of alt and ref reads
+     * @return
+     */
+    public static double hetLogLikelihood(final AFParameters parameters, final AllelicCount count) {
+        return logSumLog(hetLogLikelihood(parameters, count, AFIndicator.ALT_MINOR),
+                hetLogLikelihood(parameters, count, AFIndicator.REF_MINOR),
+                hetLogLikelihood(parameters, count, AFIndicator.OUTLIER));
+    }
+
+    // the log-likelihood of all the hets in a segment
+    public static double segmentLogLikelihood(final AFParameters parameters, final Collection<AllelicCount> counts) {
+        return counts.stream().mapToDouble(c -> hetLogLikelihood(parameters, c)).sum();
+    }
+
+
+    // the log likelihood of all segments, with modified outlier probability
+    public static double logLikelihoodWithProposedPOutlier(final AFState state, final AFData data, final double pOutlier) {
```

Comment:
it seems quite awkward to have all of these public logLikelihoodWithProposedX methods, which i guess is forced by your AdaptiveMetropolisSampler implementation (note that i haven't gone through that code carefully yet at this point).  i think it would be cleaner to have just have one exposed method that calculates the likelihood of a state, if possible.

maybe one possibility is to have AdaptiveMetropolisSampler.proposalLogLikelihood take proposed and initial states as parameters.  the responsibility for building the proposed states would then lie with the classes in AFSamplers.